### PR TITLE
No Fee State Updated

### PR DESCRIPTION
**Pull request by aparthibanpaypal**

### CODE REVIEW

File: UITests/TransferMethod/SelectTransferMethodTypeTests.swift [hunk 1]
```diff
@@ -5,7 +5,7 @@ class SelectTransferMethodTypeTests: BaseTests {
     let bankAccount = NSPredicate(format: "label CONTAINS[c] 'Bank Account'")
     let debitCard = NSPredicate(format: "label CONTAINS[c] 'Debit Card'")
     let processingTime = NSPredicate(format: "label CONTAINS[c] '\u{2022}'")
-    let transactionFee = NSPredicate(format: "label CONTAINS[c] 'fee'")
+    let transactionFee = NSPredicate(format: "label CONTAINS[c] 'No Fee'")
```
Comment (shanavas123) [inline UITests/TransferMethod/SelectTransferMethodTypeTests.swift:8]: No Fee , fee is lower cause.

Comment (aparthibanpaypal) [reply UITests/TransferMethod/SelectTransferMethodTypeTests.swift]: Fixed it. Thanks

**Review by SaiNarendran — DISMISSED** [review]

**Review by shanavas123 — DISMISSED** [review]

**Review by shanavas123 — APPROVED** [review]

**Review by SaiNarendran — APPROVED** [review]

File: Common/Sources/Extensions/HyperwalletFee.swift [hunk 2]
```diff
@@ -36,6 +36,8 @@ public extension HyperwalletFee {
             description = FeeTypes.flat(flatFee).feeDescription()
         } else if let percentFee = percentFee {
             description = FeeTypes.percent(percentFee).feeDescription()
+        } else {
+            description = FeeTypes.noFee.feeDescription()
         }
```
Comment (jsaini-hw) [inline Common/Sources/Extensions/HyperwalletFee.swift:41]: there will never be this case so no need of adding it. If there's no fee present then this func will never be called.

File: Common/Sources/Extensions/HyperwalletFee.swift [hunk 3]
```diff
@@ -108,28 +129,75 @@ public extension HyperwalletFee {
...
-                    format: feeFormat, currencySymbol, flatValue, percentValue, min)
-            } else if min == nil, let max = max, let flatValue = flatValue,
-                let percentValue = percentValue, let currencySymbol = currencySymbol(currency: currency) {
-                feeFormat = "fee_mix_only_max_formatter".localized()
-                description = String(
-                    format: feeFormat, currencySymbol, flatValue, percentValue, max)
+            
+            if flatValue?.formatAmountToDouble() == 0 && percentValue?.formatAmountToDouble() == 0 {
+                return noFeeDescription()
+            }
```
Comment (jsaini-hw) [inline Common/Sources/Extensions/HyperwalletFee.swift:135]: you can simplify the entire logic by adding else conditions to this if like 
`if flatValue?.formatAmountToDouble() == 0 && percentValue?.formatAmountToDouble() == 0 {
                return noFeeDescription()
            } else if flatValue?.formatAmountToDouble() == 0 {
                return percentFeeDescription(percentFee)
            } else if percentValue?.formatAmountToDouble() == 0 {
                return flatFeeDescription(flatFee)
            }` and then keep the existing logic of

`if let min = min, let max = max, let flatValue = flatValue,
                let percentValue = percentValue, let currencySymbol = currencySymbol(currency: currency) {
                feeFormat = "fee_mix_formatter".localized()
                description = String(
                    format: feeFormat, currencySymbol, flatValue, percentValue, min, max)
            } else if let min = min, max == nil, let flatValue = flatValue,
                let percentValue = percentValue, let currencySymbol = currencySymbol(currency: currency) {
                feeFormat = "fee_mix_only_min_formatter".localized()
                description = String(
                    format: feeFormat, currencySymbol, flatValue, percentValue, min)
            } else if min == nil, let max = max, let flatValue = flatValue,
                let percentValue = percentValue, let currencySymbol = currencySymbol(currency: currency) {
                feeFormat = "fee_mix_only_max_formatter".localized()
                description = String(
                    format: feeFormat, currencySymbol, flatValue, percentValue, max) else {...}`

Comment (aparthibanpaypal) [reply Common/Sources/Extensions/HyperwalletFee.swift:135]: Yes that is great.

File: Common/Sources/Extensions/HyperwalletFee.swift (hunk 2 above)
Comment (aparthibanpaypal) [reply Common/Sources/Extensions/HyperwalletFee.swift]: Updated

**Review by SaiNarendran — APPROVED** [review]

**Review by shanavas123 — APPROVED** [review]

File: Common/Sources/Extensions/HyperwalletFee.swift [hunk 4]
```diff
@@ -36,7 +36,7 @@ public extension HyperwalletFee {
             description = FeeTypes.flat(flatFee).feeDescription()
         } else if let percentFee = percentFee {
             description = FeeTypes.percent(percentFee).feeDescription()
-        }
+        }
```
Comment (jsaini-hw) [inline Common/Sources/Extensions/HyperwalletFee.swift:39]: why space? Didn't this give swiftlint warning?

File: Common/Sources/Extensions/HyperwalletFee.swift [hunk 5]
```diff
@@ -95,6 +110,8 @@ public extension HyperwalletFee {
                 if let value = value {
                     feeFormat = "fee_percent_no_min_and_max_formatter".localized()
                     description = String(format: feeFormat, value)
+                } else {
+                    description = FeeTypes.noFee.feeDescription()
```
Comment (jsaini-hw) [inline Common/Sources/Extensions/HyperwalletFee.swift:114]: why this one uses `FeeTypes.noFee.feeDescription()` and the in flatFeeDescription it's `else {
                description = noFeeDescription()
            }` Just keep it consistent

File: Common/Sources/Extensions/HyperwalletFee.swift (hunk 4 above)
Comment (aparthibanpaypal) [reply Common/Sources/Extensions/HyperwalletFee.swift]: Removed space.

File: Common/Sources/Extensions/HyperwalletFee.swift (hunk 5 above)
Comment (aparthibanpaypal) [reply Common/Sources/Extensions/HyperwalletFee.swift:114]: Updated with `FeeTypes.noFee.feeDescription()` to have consistency

**Review by jsaini-hw — APPROVED** [review]

**Review by malaw-hw — APPROVED** [review]

**Comment by jsaini-hw** [comment]
Flaky tests failing so merging using admin rights

**Pull request merged**